plugins/modules/proxmox_snap.py

```diff
@@ -74,6 +74,13 @@
       - Name of the snapshot that has to be created/deleted/restored.
     default: 'ansible_snap'
     type: str
+  retention:
+    description:
+      - remove old snapshots if there are more than O(retention) snapshots.
+      - If O(retention) is set to V(0), all snapshots will be kept.
```

Comment:
The behavior should be properly documented:
```suggestion
      - If O(retention) is set to V(0), all snapshots will be kept.
      - This is only used when O(state=present) and when an actual snapshot is created.
        If no snapshot is created, all existing snapshots will be kept.
```